utils/network_test.go

```diff
@@ -25,70 +28,140 @@ func TestNetwork(t *testing.T) {
+			t.Run(fmt.Sprintf("Network %v", tc.network), func(t *testing.T) {
+				assert.Equal(t, tc.feederURL, tc.network.FeederURL)
+				assert.Equal(t, tc.gatewayURL, tc.network.GatewayURL)
+			})
 		}
 	})
+
 	t.Run("chainId", func(t *testing.T) {
 		for n := range networkStrings {
 			switch n {
-			case utils.Goerli, utils.Integration:
-				assert.Equal(t, new(felt.Felt).SetBytes([]byte("SN_GOERLI")), n.ChainID())
-			case utils.Mainnet:
-				assert.Equal(t, new(felt.Felt).SetBytes([]byte("SN_MAIN")), n.ChainID())
-			case utils.Goerli2:
-				assert.Equal(t, new(felt.Felt).SetBytes([]byte("SN_GOERLI2")), n.ChainID())
+			case &utils.Goerli, &utils.Integration:
+				assert.Equal(t, new(felt.Felt).SetBytes([]byte("SN_GOERLI")), n.ChainIDFelt())
+			case &utils.Mainnet:
+				assert.Equal(t, new(felt.Felt).SetBytes([]byte("SN_MAIN")), n.ChainIDFelt())
+			case &utils.Goerli2:
+				assert.Equal(t, new(felt.Felt).SetBytes([]byte("SN_GOERLI2")), n.ChainIDFelt())
+			case &utils.Sepolia:
+				assert.Equal(t, new(felt.Felt).SetBytes([]byte("SN_SEPOLIA")), n.ChainIDFelt())
+			case &utils.SepoliaIntegration:
+				assert.Equal(t, new(felt.Felt).SetBytes([]byte("SN_INTEGRATION_SEPOLIA")), n.ChainIDFelt())
 			default:
 				assert.Fail(t, "unexpected network")
 			}
 		}
 	})
 	t.Run("default L1 chainId", func(t *testing.T) {
 		for n := range networkStrings {
-			got := n.DefaultL1ChainID()
+			got := n.L1ChainID
 			switch n {
-			case utils.Mainnet:
+			case &utils.Mainnet:
 				assert.Equal(t, big.NewInt(1), got)
-			case utils.Goerli, utils.Goerli2, utils.Integration:
+			case &utils.Goerli, &utils.Goerli2, &utils.Integration:
 				assert.Equal(t, big.NewInt(5), got)
+			case &utils.Sepolia, &utils.SepoliaIntegration:
+				assert.Equal(t, big.NewInt(11155111), got)
 			default:
 				assert.Fail(t, "unexpected network")
 			}
 		}
 	})
 }
 
-//nolint:dupl // see comment in utils/log_test.go
 func TestNetworkSet(t *testing.T) {
 	for network, str := range networkStrings {
 		t.Run("network "+str, func(t *testing.T) {
 			n := new(utils.Network)
 			require.NoError(t, n.Set(str))
-			assert.Equal(t, network, *n)
+			assert.Equal(t, network, n)
 		})
 		uppercase := strings.ToUpper(str)
 		t.Run("network "+uppercase, func(t *testing.T) {
 			n := new(utils.Network)
 			require.NoError(t, n.Set(uppercase))
-			assert.Equal(t, network, *n)
+			assert.Equal(t, network, n)
 		})
 	}
 
-	t.Run("unknown network", func(t *testing.T) {
+	t.Run("custom network - success", func(t *testing.T) {
 		n := new(utils.Network)
-		require.ErrorIs(t, n.Set("blah"), utils.ErrUnknownNetwork)
+		networkJSON := `{
+				"name": "custom",
+				"feeder_url": "baseURL/feeder_gateway/",
+				"gateway_url": "baseURL/gateway",
+				"chain_id": "SN_CUSTOM",
+				"l1_chain_id": 123,
+				"core_contract_address": "0x1",
+				"block_hash_meta_info": {
+					"first_07_block": 1,
+					"unverifiable_range": [2, 3],
+					"fallback_sequencer_address": "0x0"
+				}
```

Comment:
I don't think we should support these for the custom networks, they are only there because SW operated networks have a lot of baggage.